### DISCUSSION TITLE
URL query string

### DISCUSSION THREAD

**Discussion by stusemple** [JoTec2002/InstagramToMealie#2] [Ideas]
Would it be possible to allow url query strings that would allow some automation to take place?

I think from my research that the text box would need a ‘value’ html property  that would then allow it to be targeted in the url

what I am trying to do is create an iOS shortcut that would allow me to share the instagram link using a shortcut the share sheet that would open the webpage with the url pre populated 
Eg instagramtomealie.my server.com?textbox=instagramurl

**Comment by JoTec2002** [comment]
I impemented it in the form instagramtomealie.my-server.com?url=instagramurl it should be available in the `latest` docker image.
Hope this is what you wanted

**Comment by Scot-Survivor** [comment]
> Would it be possible to allow url query strings that would allow some automation to take place?
> 
> I think from my research that the text box would need a ‘value’ html property that would then allow it to be targeted in the url
> 
> what I am trying to do is create an iOS shortcut that would allow me to share the instagram link using a shortcut the share sheet that would open the webpage with the url pre populated Eg instagramtomealie.my server.com?textbox=instagramurl

Hi do you have your iOS shortcut on this? I can't seem to get it to encode properly

**Reply by joestump** [reply]
@Scot-Survivor I posted one in #4 - it falls back to regular Mealie URL import for non-IG URLs.

**Reply by stusemple** [reply]
Hi sorry for the delay. Just tested this and it doesn’t seem to be working for me unfortunately.

**Reply by joestump** [reply]
@stusemple here's the one that both my partner and I use: https://www.icloud.com/shortcuts/4728f51e401f4901ad74748d6157fbb3 

Post a screenshot or error and I'll take a look if this one doesn't work. You'll need 3x things:

1. InstagramToMealie URL
2. Mealie URL
3. Mealie Bearer API Token

Hope that helps!

**Reply by stusemple** [reply]
sorry for the delay. I found out it was working, it was just that I was being impatient. i needed to wait a while for the server to process the information.

Thanks for sharing your shortcut, that is ace
